src/utils/type.ts

```diff
@@ -6,3 +6,21 @@ export type Nullish = null | undefined;
 export type Prettify<T> = {
   [K in keyof T]: T[K];
 } & {};
+
+type KebabCaseHelper<S extends string> = S extends `${infer First}${infer Rest}`
+  ? First extends Lowercase<First> | '-' | '_' | ' '
+    ? `${First}${KebabCaseHelper<Rest>}`
+    : `-${Lowercase<First>}${KebabCaseHelper<Rest>}`
+  : '';
+
+export type NormalizeChars<S extends string> = S extends `${infer Head}${infer Tail}`
+  ? Head extends '_' | ' '
+    ? `-${NormalizeChars<Tail>}`
+    : `${Head}${NormalizeChars<Tail>}`
+  : S;
+
+export type KebabCase<S extends string> = S extends `${infer First}${infer Rest}`
```

Comment:
I'm not sure to understand what this type is for.
If you look at this typescript playground, I can set any kind of case without getting an error:
[Playground Link](https://www.typescriptlang.org/play/?#code/C4TwDgpgBA0hBGBDeBhRBnCAJCAbSATgDwDKUEAHsBAHYAm6U6wBAljQOYB8UAvFGUrV6jAAYASAN7sAZhAJQAYqwLMAvlNnyoAJQjrRAKChQA-EpXNyVWgygAZAPYB3eQGMMEIstXAeAHygAcgBaIKhAoIB9cMjg4xMzKAlJH3UpOCRUTxx8eSI9Zi41I0SoAC5kkKknVwIPTG9LPw1JTOQ0TFzCAv0W0orgoIBuQ0NQSCgAOUcCAFtEXFYALwgUAAtEVVJrYTtmNk4efkEbEWTNGjkFHEQ6Vq0FABVEVlwShPNbul3bRmjYsF4mVzKJqpIZvNFis1ptti83sUBiZKilvq1IQslqsNlt0EQEbgkQlKiRRuNwNB2tlGqc9owDuxuHwBL9zilHhZfA8rtpCsAPiZzGlgGy7LV3J4mr4uAkhRdUs1WtTOtg8D1+cSyqiai5JY0RcUMggOjl1flNYLBmSxjIAK40NzAViOGhQMCHYAACnQlRVUsZRwAlFBJAk3K70I5cBAAHS4RwcH1BwxqMYe9jegBEHEc9jtcyzKYzNGzuaiuALRcMJbLjhClcLKbGQA)